godot-ffi/src/godot_ffi.rs

```diff
@@ -94,6 +94,59 @@ pub unsafe trait GodotFfi {
     unsafe fn move_return_ptr(self, dst: sys::GDExtensionTypePtr, call_type: PtrcallType);
 }
 
+/// Marks a type as having a nullable counterpart in Godot.
+///
+/// # Safety
+///
+/// The type has to have a pointer sized counter part in Godot which needs to be nullable.
+/// So far only types that inherit from Object are nullable.
+pub unsafe trait GodotNullablePtrSized: GodotFfi {}
+
+unsafe impl<T> GodotFfi for Option<T>
+where
+    T: GodotNullablePtrSized,
+{
+    fn sys(&self) -> sys::GDExtensionTypePtr {
+        match self {
+            Some(value) => value.sys(),
+            None => {
+                let nil_ptr = ptr::null_mut();
+                let new_nil = interface_fn!(variant_new_nil);
+
+                unsafe {
+                    new_nil(nil_ptr);
+                }
+
+                nil_ptr as sys::GDExtensionTypePtr
+            }
+        }
+    }
+
+    unsafe fn from_sys(ptr: sys::GDExtensionTypePtr) -> Self {
+        ptr_then(ptr, |ptr| T::from_sys(ptr))
+    }
+
+    unsafe fn from_sys_init(init_fn: impl FnOnce(sys::GDExtensionTypePtr)) -> Self {
+        let mut raw = std::mem::MaybeUninit::uninit();
+        init_fn(raw.as_mut_ptr() as sys::GDExtensionTypePtr);
+
+        Self::from_sys(raw.assume_init())
+    }
+
+    unsafe fn from_arg_ptr(ptr: sys::GDExtensionTypePtr, call_type: PtrcallType) -> Self {
+        ptr_then(ptr, |ptr| T::from_arg_ptr(ptr, call_type))
+    }
+
+    unsafe fn move_return_ptr(self, ptr: sys::GDExtensionTypePtr, call_type: PtrcallType) {
+        match self {
+            Some(value) => value.move_return_ptr(ptr, call_type),
+            None => {
+                interface_fn!(variant_new_nil)(ptr as sys::GDExtensionVariantPtr);
```

Comment:
do we even need to do anything here when it's `None`?

Reply:
Appears to be unnecessary.